File: Codigo/Mobile/lib/usuario.dart
class Usuario {
  int _id;
  String _nome;
  bool _funcionario;
  String _senha;
  String _login;

  Usuario(this._id, this._nome, this._funcionario, this._senha, this._login);

  int get id => _id;
  set id(int value) => _id = value;

  String get nome => _nome;
  set nome(String value) => _nome = value;

  bool get funcionario => _funcionario;
  set funcionario(bool value) => _funcionario = value;

  String get senha => _senha;
  set senha(String value) => _senha = value;

  String get login => this._login;
  set login(String value) => _login = value;
}

File: Codigo/Desktop/ProjetoBandejao/ProjetoBandejao/Models/Usuario.cs
using System;
using System.Text.Json.Serialization;

namespace ProjetoBandejao.Models
{
    public class Usuario
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [JsonPropertyName("nome")]
        public string Nome { get; set; } = string.Empty;

        [JsonPropertyName("login")]
        public string Login { get; set; } = string.Empty;

        [JsonPropertyName("senhaHash")]
        public string Senha { get; set; } = string.Empty;

        [JsonPropertyName("funcionario")]
        public bool Funcionario { get; set; }

        [JsonPropertyName("emailConfirmado")]
        public bool EmailConfirmado { get; set; }

        public string Cargo => Funcionario ? "Administrador / Funcionário" : "Cliente / Aluno";
        public string StatusFormatado => EmailConfirmado ? "Ativo" : "Confirmação Pendente";
    }
}
